orchestrators/Orchestrator.py

```diff
@@ -0,0 +1,443 @@
+from abc import ABC, abstractmethod
+import os
+import json
+import logging
+import requests
+import copy
+from dotenv import load_dotenv
+
+class Orchestrator(ABC):
+    load_dotenv()
+
+    DEBUG = os.environ.get("DEBUG", "false")
+    DEBUG_LOGGING = DEBUG.lower() == "true"
+
+    @abstractmethod
+    def conversation_with_data(self, request_body):
+        pass
+
+    @abstractmethod
+    def conversation_without_data(self, request_body):
+        pass
+
+    # Initialize search variables
```

Comment:
If any of these settings can/will be used in other code it might make sense to abstract them into individual config class and then import them here, one for each domain, e.g., AzureOpenAIConfig.py etc.